src/main/java/com/three_tech_solutions/slot_app/services/implementations/StudentServiceImpl.java

```diff
@@ -0,0 +1,95 @@
+    private final UserService userService;
+    private static final Set<String> ALLOWED_PLAN_TYPES = Set.of("Día específico", "Del 1 al 10");
+
+    //constructor
+    public StudentServiceImpl(StudentRepository studentRepository, StudentMapper studentMapper,
+                              PlanTypeService planTypeService, PlanService planService, UserService userService) {
+
+        this.studentRepository = studentRepository;
+        this.studentMapper = studentMapper;
+        this.planTypeService = planTypeService;
+        this.planService = planService;
+        this.userService = userService;
+    }
+
+    @Override
+    public StudentResponseDTO createStudent(StudentRequestDTO studentDTO) {
+        // Lógica de negocio: validaciones, cálculos, guardarVALIDACIONES, Existe el alumno?, Tiene todos los datos?
+
+        //La fecha de admisión en caso de no recibirse, debe ser por defecto la fecha actual
+        if (studentDTO.getAdmissionDate() == null) {
+            studentDTO.setAdmissionDate(LocalDate.now());
+        }
+        // Buscar el tipo de plan por ID
+        PlanType planType = planTypeService.getById(studentDTO.getPlanTypeId())
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "PlanType no encontrado"));
+        // Validar que el tipo de plan sea permitido
+        if (!ALLOWED_PLAN_TYPES.contains(planType.getName())) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Tipo de plan no permitido.");
+        }
+        // Validar día de pago si es "Día específico"
+        if (planType.getName().equalsIgnoreCase("Día específico")) {
+            if (studentDTO.getPaymentDay() <= 0 || studentDTO.getPaymentDay() > 31) {
+                throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "El día de pago debe ser entre 1 y 31.");
+            }
+        } else {
+        // Si no es "Día específico", el día de pago debe quedar en null
+        studentDTO.setPaymentDay( null);
+    }
+        //Las clases extra tiene que ser un valor opcional (obligatorio en caso de que el tipo de pago sea del 1-10 y la fecha actual sea mayor al 10 del mes), pero si se ingresa debe ser mayor a 10 y menor a 20.
+        //El día de pago debe ser un dato opcional (obligatorio en caso de que el tipo de pago sea dia específico) y debe ser mayor o igual a 1 y menor o igual a 31
+        // Crear y guardar el Plan
+        Plan plan = new Plan();
+        plan.setId(UUID.randomUUID());
+        plan.setClassesPerWeek(studentDTO.getClassesPerWeek());
+        plan.setPaymentDay(studentDTO.getPaymentDay());
+        plan.setPaymentType(planType);
+
+        planService.save(plan);
+
+        //Al registrar el alumno, debe vincularse con el usuario (profesional logueado)
+        User user = userService.getById(studentDTO.getUserId())
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "Usuario del profesinal no encontrado"));
```

Comment:
La logica de lanzar una excepcion cuando el usuario no existe, deberia ser parte del userService y no deberia ser logica del StudentService. Nosotros aca debeiramos hacer algo como esto

```suggestion
        User user = userService.getByIdOrThrowException(studentDTO.getUserId());
```

y el userService se encarga de la parte de lanzar la exception si el user no existe. El metodo `getByIdOrThrowException()` devuelve SI o SI un usuario, no retorna un `Optional<User>`